src/metabase/search/filter.clj

```diff
@@ -77,6 +78,31 @@
 
 (defmethod where-clause* ::list [_ k v] [:in k v])
 
+(defn personal-collections-where-clause
+  "Build a clause limiting the entries to those (not) within or within personal collections, if relevant.
+  WARNING: this method queries the appdb, and its approach will get very slow when there are many users!"
+  [{filter-type :filter-items-in-personal-collection} collection-id-col]
+  (when filter-type
+    (let [parent-ids     (t2/select-pks-vec :model/Collection :personal_owner_id [:not= nil])
+          child-patterns (for [id parent-ids] (format "/%d/%%" id))]
+      (case filter-type
+        "only"
+        `[:or
```

Comment:
Haven't really used quasi quotes like this before, but seems like a nice way to splice list expressions deeply in nested collections.

Reply:
😮

Reply:
It's surprising, but also readable, so I'm kind of ambivalent about it 😁